xdsl/backend/assembly_printer.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+from xdsl.dialects.builtin import ModuleOp, StringAttr
+from xdsl.utils.base_printer import BasePrinter
+
+
+class AssemblyPrinter(BasePrinter):
+    @staticmethod
+    def append_comment(line: str, comment: StringAttr | None) -> str:
+        if comment is None:
+            return line
+
+        padding = " " * max(0, 48 - len(line))
+
+        return f"{line}{padding} # {comment.data}"
+
+    @staticmethod
+    def assembly_line(
+        name: str,
+        arg_str: str,
+        comment: StringAttr | None = None,
+        is_indented: bool = True,
+    ) -> str:
+        code = "    " if is_indented else ""
+        code += name
+        if arg_str:
+            code += f" {arg_str}"
+        code = AssemblyPrinter.append_comment(code, comment)
+        return code
+
+    def print_module(self, module: ModuleOp) -> None:
+        for op in module.body.walk():
+            assert isinstance(op, AssemblyPrintable), f"{op}"
+            op.print_assembly(self)
+
+
+class AssemblyPrintable(ABC):
```

Comment:
```suggestion
class AssemblyPrintable(Operation, ABC):
```
Is it worth adding this to agree with the docstring below (I'm not sure if this is good practice in general).